src/wallet/wallet.ts

```diff
@@ -354,44 +679,434 @@ class HathorWalletServiceWallet extends EventEmitter {
     return address.toString();
   }
 
-  getCurrentAddress() {
-    throw new Error('Not implemented.');
+  /**
+   * Get the current address to be used
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  getCurrentAddress({ markAsUsed = false } = {}): AddressInfoObject {
+    const newAddressesLen = this.newAddresses.length;
+    if (this.indexToUse > newAddressesLen - 1) {
+      const addressInfo = this.newAddresses[newAddressesLen - 1];
+      return {...addressInfo, error: 'GAP_LIMIT_REACHED'};
```

Comment:
Shouldn't this throw instead of returning an object?

Reply:
At first I agreed with you but I was changing this part of the code to throw the error and I had second thoughts. The user that calls the method `getCurrentAddress` wants an address to use. It doesn't matter which one, we just need to warn this user that the gap limit was reached and from now on the address will always be the same until one is used. This warning could come from the error thrown but how would I return the address to the user? We could assume that the user already has a previous address but he might not have it, so even when it reaches the gap limit, I need to return the address in this method, that's why I think throwing an error might not be the best approach here. What do you think?

Reply:
I see, so the main point here is that this is not an error, this is expected behavior -- we reached the gap limit

I would just change the attribute `error` to something like `info`. What do you think?

Reply:
I agree! Changed to info